### PR TITLE
Add initial XRT runtime cmake integration

### PR DESCRIPTION
**Pull request by nirvedhmeshram**
This just has a small test to verify that we can access the XRT API

### CODE REVIEW

**Comment by nirvedhmeshram** [comment]
I am looking for suggestions on how to improve this and then we can merge this as a starting point for the runtime.

**Review by MaheshRavishankar — COMMENTED** [review]
Add new line at end of file. Overall looks fine to me. Stella is the one who really know this.

File: iree_compiler_plugin_group.cmake
Comment (stellaraccident) [inline iree_compiler_plugin_group.cmake:10]: Ideally we aren't bundling this into the compiler plugin system but I know why you are (it is easy/accessible).

I need to do a bit more cmake hacking to make this separable in a similar way that the compiler side is (basically, we should be able to build compiler and runtime separately).

Ultimately, we will want to use the `IREE_EXTERNAL_TOOLING_MODULES` facility (e.g. https://github.com/openxla/openxla-nvgpu/blob/main/CMakeLists.txt#L93) to provide additional features. This plumbing has been used a few times but it may need a bit more work. I can look at that after you land.

**Review by stellaraccident — APPROVED** [review]
I left a number of comments, mostly for education. Please address the lint, but we can fix forward on the others if you want to iterate on landed patches.

**Comment by nirvedhmeshram** [comment]
> I left a number of comments, mostly for education. Please address the lint, but we can fix forward on the others if you want to iterate on landed patches.

@stellaraccident , Thanks I should be able to and would like to address all of your comments before merging, will get to it in a bit though after I make some progress (or get blocked on) using the runtime API

**Comment by stellaraccident** [comment]
Cool, dm me for a follow-up if you'd like another round of review.

File: runtime/plugins/XRT/runtime/CMakeLists.txt
Comment (nirvedhmeshram) [reply runtime/plugins/XRT/runtime/CMakeLists.txt:20]: Thanks, they do have find_packages but the paths I get from it have not been useful to replace the headers/libs I am using here. e.g this cmake
```
find_package(XRT REQUIRED)
message(${XRT_INCLUDE_DIRS})
message(${XRT_CORE_LIBRARIES})
message(${XRT_LINK_DIRS})
```
is giving this output
```
-- Registering static linked compiler plugin 'amd_aie' (iree::target::amd-aie::registration)
-- Found XRT: /home/nirvedh/XRT/build/Debug (found version "2.17.0")
/home/xrt/include
XRT::xrt_coreXRT::xrt_coreutil
/home/xrt/lib
```
so the paths it is giving are not actually real.

File: runtime/plugins/XRT/runtime/CMakeLists.txt
Comment (nirvedhmeshram) [reply runtime/plugins/XRT/runtime/CMakeLists.txt]: We will be working on packaging in the following PRs

File: runtime/plugins/XRT/runtime/CMakeLists.txt
Comment (nirvedhmeshram) [reply runtime/plugins/XRT/runtime/CMakeLists.txt]: this is solved, had to install the package at /opt/xilinix, I have updated the readme to reflect this

File: iree_compiler_plugin_group.cmake
Comment (nirvedhmeshram) [reply iree_compiler_plugin_group.cmake:10]: we will be working on improving this in the following PRs

**Pull request merged**
